Application/EdFi.Admin.DataAccess/Contexts/UsersContextFactory.cs

```diff
@@ -28,16 +29,44 @@ public UsersContextFactory(IAdminDatabaseConnectionStringProvider connectionStri
             _connectionStringsProvider = Preconditions.ThrowIfNull(connectionStringsProvider, nameof(connectionStringsProvider));
             _databaseEngine =  Preconditions.ThrowIfNull(databaseEngine, nameof(databaseEngine));
         }
+        
+        public Type GetUsersContextType()
+        {
+            if (_usersContextTypeByDatabaseEngine.TryGetValue(_databaseEngine, out Type contextType))
+            {
+                return contextType;
+            }
+
+            throw new InvalidOperationException(
+                $"No UsersContext defined for database type {_databaseEngine.DisplayName}");
+        } 
 
         public IUsersContext CreateContext()
         {
-            if (_usersContextTypeByDatabaseEngine.TryGetValue(_databaseEngine, out Type contextType))
+            if (_databaseEngine == DatabaseEngine.SqlServer)
+            {
+                return Activator.CreateInstance(
+                        GetUsersContextType(),
+                        new DbContextOptionsBuilder<SqlServerUsersContext>()
+                            .UseLazyLoadingProxies()
+                            .UseSqlServer(_connectionStringsProvider.GetConnectionString())
+                            .Options) as
+                    IUsersContext;
+            }
+
+            if (_databaseEngine == DatabaseEngine.Postgres)
             {
-                return Activator.CreateInstance(contextType, _connectionStringsProvider.GetConnectionString()) as IUsersContext;
+                return Activator.CreateInstance(
+                        GetUsersContextType(),
+                        new DbContextOptionsBuilder<PostgresUsersContext>()
+                            .UseLazyLoadingProxies()
+                            .UseNpgsql(_connectionStringsProvider.GetConnectionString())
+                            .Options) as
+                    IUsersContext;
             }
 
             throw new InvalidOperationException(
-                $"Cannot create an IUsersContext for database type {_databaseEngine.DisplayName}");
+                $"Cannot create an SecurityContext for database type {_databaseEngine.DisplayName}");
```

Comment:
The method is still creating a UserContext

Reply:
This is reverted